File: src/storage/src/main/java/sk/kasv/infinity/model/Subscription.java
package sk.kasv.infinity.model;

import java.util.Date;

public class Subscription {

  private Long id;
  private Long userId;
  private Date startDate;
  private Date endDate;

  public Subscription(Long id, Long userId, Date startDate, Date endDate) {
    this.id = id;
    this.userId = userId;
    this.startDate = startDate;
    this.endDate = endDate;
  }

  public Subscription() {

  }

  public Long getId() {
    return id;
  }

  public Long getUserId() {
    return userId;
  }

  public Date getStartDate() {
    return startDate;
  }

  public Date getEndDate() {
    return endDate;
  }

  public void setId(Long id) {
    this.id = id;
  }

  public void setUserId(Long userId) {
    this.userId = userId;
  }

  public void setStartDate(Date startDate) {
    this.startDate = startDate;
  }

  public void setEndDate(Date endDate) {
    this.endDate = endDate;
  }

  @Override
  public String toString() {
    final StringBuilder sb = new StringBuilder("Subscription{");
    sb.append("id=").append(id);
    sb.append(", userId=").append(userId);
    sb.append(", startDate=").append(startDate);
    sb.append(", endDate=").append(endDate);
    sb.append('}');
    return sb.toString();
  }
}

File: src/entities/subscription.ts
export class Subscription {
    public static clone(s: Subscription):Subscription{
        return new Subscription(s.userId, s.startDate, s.endDate, s.id);
    }

    constructor(
        public userId: number,
        public startDate: Date,
        public endDate: Date,
        public id?: number,
    ){}

    public toString():string {
        return `id: ${this.id}, userId: ${this.userId}, startDate: ${this.startDate}, endDate: ${this.endDate}`;

    }
}